agent/app/agent_windows_test.go

```diff
@@ -81,3 +88,242 @@ func TestDoStartHappyPath(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	stateManager := statemanager_mocks.NewMockStateManager(ctrl)
+	taskEngine := engine.NewMockTaskEngine(ctrl)
+	defer ctrl.Finish()
+
+	taskEngine.EXPECT().Disable()
+	stateManager.EXPECT().ForceSave()
+
+	agent := &mockAgent{}
+
+	done := make(chan struct{})
+	defer func() { done <- struct{}{} }()
+	startFunc := func() int {
+		go agent.terminationHandler(stateManager, taskEngine)
+		<-done // block until after the test ends so that we can test that runAgent returns when cancelled
+		return 0
+	}
+	agent.startFunc = startFunc
+	handler := &handler{agent}
+	ctx, cancel := context.WithCancel(context.TODO())
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		handler.runAgent(ctx)
+		wg.Done()
+	}()
+	time.Sleep(time.Second) // give startFunc enough time to actually call the termination handler
+	cancel()
+	wg.Wait()
+}
+
+func TestHandler_HandleWindowsRequests_StopService(t *testing.T) {
+	requests := make(chan svc.ChangeRequest)
+	responses := make(chan svc.Status)
+
+	handler := &handler{}
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+
+	go func() {
+		handler.handleWindowsRequests(context.TODO(), requests, responses)
+		wg.Done()
+	}()
+
+	go func() {
+		resp := <-responses
+		assert.Equal(t, svc.StartPending, resp.State, "Send StartPending immediately")
+		resp = <-responses
+		assert.Equal(t, svc.Running, resp.State, "Send Running after StartPending")
+		assert.Equal(t, svc.AcceptStop|svc.AcceptShutdown, resp.Accepts, "Accept stop & shutdown")
+		requests <- svc.ChangeRequest{Cmd: svc.Interrogate, CurrentStatus: svc.Status{State: svc.Running}}
+		resp = <-responses
+		assert.Equal(t, svc.Running, resp.State, "Send Running after Interrogate")
+		requests <- svc.ChangeRequest{Cmd: svc.Stop}
+		resp = <-responses
+		assert.Equal(t, svc.StopPending, resp.State, "Send StopPending after Stop")
+		wg.Done()
+	}()
+
+	wg.Wait()
+}
+
+func TestHandler_HandleWindowsRequests_Cancel(t *testing.T) {
+	requests := make(chan svc.ChangeRequest)
+	responses := make(chan svc.Status)
+
+	handler := &handler{}
+	ctx, cancel := context.WithCancel(context.TODO())
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+
+	go func() {
+		handler.handleWindowsRequests(ctx, requests, responses)
+		wg.Done()
+	}()
+
+	go func() {
+		resp := <-responses
+		assert.Equal(t, svc.StartPending, resp.State, "Send StartPending immediately")
+		resp = <-responses
+		assert.Equal(t, svc.Running, resp.State, "Send Running after StartPending")
+		assert.Equal(t, svc.AcceptStop|svc.AcceptShutdown, resp.Accepts, "Accept stop & shutdown")
+		requests <- svc.ChangeRequest{Cmd: svc.Interrogate, CurrentStatus: svc.Status{State: svc.Running}}
+		resp = <-responses
+		assert.Equal(t, svc.Running, resp.State, "Send Running after Interrogate")
+		cancel()
+		resp = <-responses
+		assert.Equal(t, svc.StopPending, resp.State, "Send StopPending after Cancel")
+		wg.Done()
+	}()
+
+	wg.Wait()
+}
+
+func TestHandler_Execute_WindowsStops(t *testing.T) {
```

Comment:
Unit tests are run with a 25s timeout right now.  This completes in about a second.  Our unit tests aren't really able to simulate the worst-case right now, so I'd rather not add an assert that won't tell us anything or try to make the unit test different.  Instead, I think this can be addressed in a higher-level acceptance test (probably outside the agent itself).